exe/Daemon.hs

```diff
@@ -238,6 +239,28 @@ send id (Api.SendRequest expressions) = do
         Nothing -> throw $ MachineError id AckTimeout
 
 
+-- | Give the HTML specified by a machine (via defining `(get-html)`).
+frontend :: MachineId -> [Text] -> Daemon Api.HtmlText
+frontend id path = do
+    m <- ensureMachineLoaded id
+    bumpPolling id
+    case runIdentity $ interpret "[frontend]" "(get-html)" (machineState m) of
+        Left err -> throw $ MachineError id (InvalidInput err)
+        Right (String v) -> do
+            logInfo "Handled frontend request" [("machine-id", getMachineId id)]
+            case Ipfs.cidFromText v of
+                Left _ -> throw $ MachineError id
+                    (InvalidInput $ toLangError $ OtherError "get-html returned invalid CID")
+                Right v' -> do
+                    let ipfsPath = show v' <> "/" <> T.intercalate "/" path
+                    logInfo "Query HTML" [ ("CID", show v')
+                                         , ("machine-id", getMachineId id)
+                                         , ("ipfs-path", ipfsPath)
+                                         ]
+                    val <- Ipfs.ipfsHttpGet' "cat" [("arg", ipfsPath)]
+                    pure $ Api.HtmlText $ toS val
+        Right _ -> throw $ MachineError id (DaemonError "get-html returned non-string")
```

Comment:
I don't think the error message should assume `get-html` is a function.

Reply:
Fixed